### PR TITLE
[Qwen3-Omni] vectorize per-row Python loops in talker AR hot path

### PR DESCRIPTION
**Pull request by yxs**
Vectorize four per-row Python loops in the talker AR hot path. Each loop collected per-row index lists in Python and then issued one batched advanced-indexing scatter — same pattern across all four sites, no numerical operation changed

2×H200, seed-tts-eval, EN VC=F, C=16, stream=True, 5R restart-per-round mean:

| run | latency_mean(s) | rtf | qps | wer_corpus | wer<50 | n>50 |
|---|---:|---:|---:|---:|---:|---:|
| main `dde550c` (baseline) | 3.013 ± 0.07 | 0.836 ± 0.020 | 5.30 ± 0.13 | 2.72 % | 2.21 % | 8 |
| + `prepare_decode_buffers` alone (5R) | **2.591 ± 0.04** | 0.725 | 6.16 | 2.76 % | 2.25 % | 8 |
| + full bundle (5R) | **2.274 ± 0.018** | **0.641 ± 0.005** | **7.02 ± 0.05** | 2.75 % | 2.24 % | 8 |

**Cumulative: latency −24.5 %, RTF −23.3 %, QPS +32.4 %. WER and n50 unchanged.**

Closes, Part of #486.

### CODE REVIEW

File: sglang_omni/model_runner/base.py [hunk 1]
```diff
@@ -232,19 +237,28 @@ def _apply_repetition_penalty(self, logits_output: Any, requests: list) -> None:
...
-            scores = torch.where(scores > 0, scores / penalty, scores * penalty)
-            logits[row_idx, idx] = scores
+            rep_rows.extend([row_idx] * len(unique))
+            rep_toks.extend(unique)
+            rep_penalties.extend([float(penalty)] * len(unique))
+        if rep_rows:
+            rows_t = torch.tensor(rep_rows, dtype=torch.long, device=device)
+            toks_t = torch.tensor(rep_toks, dtype=torch.long, device=device)
+            pens_t = torch.tensor(rep_penalties, dtype=logits.dtype, device=device)
+            scores = logits[rows_t, toks_t]
```
Comment (zhaochenyang20) [inline sglang_omni/model_runner/base.py]: this shared pre-sampler path can change token ordering when `next_token_logits` is `float16`. The previous row-local code applied the Python float `penalty` directly in `scores / penalty` or `scores * penalty`; the new code constructs `rep_penalties` as `dtype=logits.dtype` at line 249, so common values such as 1.2 are rounded to reduced precision before the divide/multiply at line 251. A local PyTorch probe reproduces the difference: with `torch.float16`, `tensor([1.0]) / 1.2` produces `0.83349609375`, while dividing by a half tensor penalty produces `0.8330078125`. This is not just a display difference; this function mutates logits immediately before `tp_worker.model_runner.sample()`, so close candidate ordering, top-p boundaries, or greedy selection can flip even though the PR claims no numerical operation changed.


Keep the vectorized coordinate collection, but perform the penalty math in at least `float32` and cast the result back to the original logits dtype only when writing it into `logits`. Add a minimal regression test with half logits, `repetition_penalty=1.2`, and positive/negative repeated tokens, comparing the vectorized function against the previous scalar formula; the positive-logit divide path should be covered because it exposes the precision difference most clearly.

```python
orig_dtype = logits.dtype
pens_t = torch.tensor(rep_penalties, dtype=torch.float32, device=device)
scores = logits[rows_t, toks_t].to(torch.float32)
scores = torch.where(scores > 0, scores / pens_t, scores * pens_t)
logits[rows_t, toks_t] = scores.to(orig_dtype)
```

**Review by zhaochenyang20 — CHANGES_REQUESTED** [review]

File: sglang_omni/model_runner/base.py (hunk 1 above)
Comment (yxs) [reply sglang_omni/model_runner/base.py]: fp32 math + cast back, regression test added. fp16 diff vs scalar 0.0039 → 0.0019

**Review by zhaochenyang20 — APPROVED** [review]
